driver-async/src/main/com/mongodb/async/client/FindIterableImpl.java

```diff
@@ -238,9 +237,9 @@ public void onResult(final List<TResult> result, final Throwable t) {
                 .limit(findOptions.getLimit())
                 .maxTime(findOptions.getMaxTime(MILLISECONDS), MILLISECONDS)
                 .maxAwaitTime(findOptions.getMaxAwaitTime(MILLISECONDS), MILLISECONDS)
-                .modifiers(toBsonDocument(findOptions.getModifiers()))
-                .projection(toBsonDocument(findOptions.getProjection()))
-                .sort(toBsonDocument(findOptions.getSort()))
+                .modifiers(toBsonDocument(findOptions.getModifiers(), codecRegistry))
```

Comment:
Pass documentClass to toBsonDocument method so as not to change the behavior, in case there are implementations of Bson#toBsonDocument that care about the documentClass parameter (which we may at some point in our own implementations).

Reply:
Done.